internal/provisioner/controller/gateway_test.go

```diff
@@ -1263,7 +1263,54 @@ func TestGatewayReconcile(t *testing.T) {
 					},
 				}
 				require.NoError(t, r.client.Get(context.Background(), keyFor(ds), ds))
-				assert.Contains(t, ds.Spec.Template.ObjectMeta.Annotations, "key")
+				assert.Contains(t, ds.Spec.Template.ObjectMeta.Annotations, "key", "prometheus.io/scrape", "prometheus.io/port")
+			},
+		},
+
+		"If ContourDeployment.Spec.Contour.PodAnnotations is specified, the Contour pods' have annotations for prometheus & user-defined": {
+
+			gatewayClass: reconcilableGatewayClassWithParams("gatewayclass-1", controller),
+			gatewayClassParams: &contourv1alpha1.ContourDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "projectcontour",
+					Name:      "gatewayclass-1-params",
+				},
+				Spec: contourv1alpha1.ContourDeploymentSpec{
+					Contour: &contourv1alpha1.ContourSettings{
+						PodAnnotations: map[string]string{
+							"key": "val",
+						},
+					},
+				},
+			},
+			gateway: &gatewayv1beta1.Gateway{
```

Comment:
i think your other refactor PR `makeGateway` helper could be useful here

Reply:
yes,  this pr leads to that one, i will merge & refactor this to sync with it